lib/kubernetes-deploy/kubeclient_builder.rb

```diff
@@ -116,7 +116,26 @@ def _build_kubeclient(api_version:, context:, endpoint_path: nil)
 
     def config_files
       # Split the list by colon for Linux and Mac, and semicolon for Windows.
-      ENV.fetch("KUBECONFIG", "#{Dir.home}/.kube/config").split(/[:;]/).map!(&:strip).reject(&:empty?)
+      (kubeconfig || "#{Dir.home}/.kube/config").split(/[:;]/).map!(&:strip).reject(&:empty?)
+    end
+
+    def kubeconfig
+      # Cannot return a value to ensure that the default is used over env["KUBECONFIG"] when `kubeconfig: nil` is set
+      # in tests explicitly, for the case when ENV["KUBECONFIG"] is nil, since it may be set in the test environment
+    end
```

Comment:
If this is only for tests I would rather mock it out in the tests themselves than add it here. @dturn wdyt?

Reply:
What do you think of https://github.com/powerhome/kubernetes-deploy/pull/1 ?